packages/contracts/contracts/engine/V2/GenArt721Minter_PBAB.sol

```diff
@@ -207,6 +221,14 @@ contract GenArt721Minter_PBAB is ReentrancyGuard {
             require(msg.sender == tx.origin, "No Contract Buys");
         }
 
+        address configuredCurrencyAddress = genArtCoreContract
+            .projectIdToCurrencyAddress(_projectId);
+        // validate that the currency address matches the project configured currency
+        require(
+            _currencyAddress == configuredCurrencyAddress,
+            "Currency addresses must match"
+        );
+
```

Comment:
I think if `configuredCurrencyAddress  == address(0)`, we should probably require `msg.value == _maxPricePerToken` here (since anything else would be nonsense inputs, I think?)?